Description: use qstrncpy instead of strlcpy which is not available on GNU/kFreeBSD
Author: Dmitry Shachnev <mitya57@debian.org>
Forwarded: not-yet
Last-Update: 2018-10-31

--- a/src/network/kernel/qnetworkinterface_unix.cpp
+++ b/src/network/kernel/qnetworkinterface_unix.cpp
@@ -38,6 +38,7 @@
 **
 ****************************************************************************/
 
+#include "qbytearray.h"
 #include "qset.h"
 #include "qnetworkinterface.h"
 #include "qnetworkinterface_p.h"
@@ -500,7 +501,7 @@ static QList<QNetworkInterfacePrivate *>
             iface->flags = convertFlags(ptr->ifa_flags);
             iface->hardwareAddress = iface->makeHwAddress(sdl->sdl_alen, (uchar*)LLADDR(sdl));
 
-            strlcpy(mediareq.ifm_name, ptr->ifa_name, sizeof(mediareq.ifm_name));
+            qstrncpy(mediareq.ifm_name, ptr->ifa_name, sizeof(mediareq.ifm_name));
             iface->type = probeIfType(openSocket(socket), sdl->sdl_type, &mediareq);
             iface->mtu = getMtu(socket, &req);
         }
@@ -524,7 +525,7 @@ static void getAddressExtraInfo(QNetwork
         return;
     }
 
-    strlcpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
+    qstrncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
 
     // get flags
     ifr.ifr_addr = *reinterpret_cast<struct sockaddr_in6 *>(sa);
